go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -0,0 +1,970 @@
+	}
+	klog.V(5).Infof("==>Got EndpointList %v for service %s in namespace %s\n", eps, svcName, namespace)
+
+	// Get one of the podnames that make up the service
+	for _, ep := range eps.Items {
+		klog.V(5).Infof("==>Got endpoint %v for service %s in namespace %s\n", ep, svcName, namespace)
+		for _, subset := range ep.Subsets {
+			klog.V(5).Infof("==>Got subset %v for service %s in namespace %s\n", subset, svcName, namespace)
+			for _, epAddress := range subset.Addresses {
+				klog.V(5).Infof("==>Got Address %v for service %s in namespace %s\n", epAddress, svcName, namespace)
+				svcInfo.PodName = epAddress.TargetRef.Name
+				svcInfo.PodNamespace = epAddress.TargetRef.Namespace
+				svcInfo.PodIP = epAddress.IP
+				break
+			}
+			for _, port := range subset.Ports {
+				klog.V(5).Infof("==>Got Port %v for service %s in namespace %s\n", port, svcName, namespace)
+				svcInfo.PodPort = strconv.Itoa(int(port.Port))
+				break
+			}
+		}
+	}
+	if svcInfo.PodName == "" {
+		klog.V(0).Infof("Cannot find pods in Endpoints for service %s in namespace %s", svcName, namespace)
+		err := fmt.Errorf("cannot find pods in Endpoints for service %s in namespace %s", svcName, namespace)
+		return nil, err
+	}
+
+	return svcInfo, err
+}
+
+func getPodInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, podName string, ovnNamespace string, namespace string, cmd string) (podInfo *PodInfo, err error) {
+
+	var ethName string
+
+	// Get pod with the name supplied by srcPodName
+	pod, err := coreclient.Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		klog.V(1).Infof("Pod %s in namespace %s not found\n", podName, namespace)
+		return nil, err
+	}
+
+	podInfo = &PodInfo{
+		IP: pod.Status.PodIP,
+	}
+
+	// Get the node on which the pod runs on
+	node, err := coreclient.Nodes().Get(context.TODO(), pod.Spec.NodeName, metav1.GetOptions{})
+	if err != nil {
+		klog.V(1).Infof("Pod %s in namespace %s node not found\n", podName, namespace)
+		return nil, err
+	}
+	klog.V(5).Infof("==>Got pod %s which is running on node %s\n", podName, node.Name)
+
+	podMAC, err := getPodMAC(coreclient, pod)
+	if err != nil {
+		klog.V(1).Infof("Problem obtaining Ethernet address of Pod %s in namespace %s\n", podName, namespace)
+		return nil, err
+	}
+
+	podInfo.PodName = pod.Name
+	podInfo.MAC = podMAC
+	podInfo.ContainerName = pod.Spec.Containers[0].Name
+
+	var tryJSON bool = true
+	var linkIndex int
+
+	// The interface name used depends on what network namespasce the pod uses
+	if pod.Spec.HostNetwork {
+		ethName = util.GetLegacyK8sMgmtIntfName(node.Name)
+		podInfo.HostNetwork = true
+		linkIndex = 0
+	} else {
+		ethName = "eth0"
+		podInfo.HostNetwork = false
+
+		// Find index used for pod interface
+
+		ipJaddrCmd := "Ip -j addr show " + ethName
+
+		klog.V(5).Infof("Ip -j command is %s", ipJaddrCmd)
+		ipOutput, ipError, err := execInPod(coreclient, restconfig, namespace, podName, podInfo.ContainerName, ipJaddrCmd, "")
```

Comment:
does this really work? Note the ipJaddrCmd is set to "Ip" , shouldn't it be "ip" ?

Reply:
yes with newer versions of iproute2.

Reply:
yeah, but you pass ipJaddrCmd to execInPod , so the command you pass has Ip, instead of ip, so it's not just the log statement?

Reply:
agreed, I fixed:

-               ipJaddrCmd := "Ip -j addr show " + ethName
+               ipJaddrCmd := "ip -j addr show " + ethName